### ISSUE TITLE
Random failure to inject USB power properties

### ISSUE THREAD

**Issue by f00b4r0** [acidanthera/bugtracker#1121]
Hi,

I'm running Mojave 10.14.6 (18G6020) on an ASUS Z370 Prime-A (rev1) / i3-8350k, SMBIOS iMac19,1, OC 0.6.0.

When I boot the machine, every now and then the USB ports will not deliver the extra current as configured in SSDT-USBX

Sometimes it will work correctly, sometimes the ports will be limited to the USB2 or USB3 'standard' max (500 or 900mA), even though in _all cases_ `ACPI/_SB/USBX@0` is present and seemingly correctly populated in IORegistryExplorer. When it doesn't work, in System Information -> Hardware -> USB, for each device Available Sleep Current is not displayed and Extra Operating Current is displayed but remains 0 no matter what is plugged.

I haven't been able to identify what causes this problem, it's a hit-and-miss: I have to reboot multiple times until the extra current becomes effectively available.

This problem did not occur with Clover with the exact same machine/kexts.

More interestingly, I recently tried OC on an older build: ASUS H110M-A.M2 / i3-6100, SMBIOS iMac14,2, also Mojave 10.14.6 (18G6020). This is an SMBIOS that does not require SSDT-USBX and this is where it gets interesting: with Clover, no problem, the high power properties are natively present (no extra configuration of any kind) and the USB ports can deliver extra power. However, with OC 0.6.0, they're not and high current doesn't work on the USB ports.

This may indicate that this isn't related to ACPI injection, but instead it seems there is some odd interaction between OC and the USB driver. It has the smell of a race, but I'm not sure how to further debug it. I'm happy to do so however, let me know if I can try something useful.

HTH

**Comment by Andrey1970AppleLife** [comment]
Use forums for support.

**Comment by f00b4r0** [comment]
@Andrey1970AppleLife this isn't a support request. This is a bug (or so I believe). You're welcome to ignore it, I guess.

**Comment by vit9696** [comment]
I do not believe this is an OpenCore issue, most likely some BIOS problem. On one of the boards I had to use ExitBootServices delay to workaround BIOS issues, on some other — ReleaseUsbOwnership. All in all, it is very likely a configuration issue, and supporting your board is practically out of the scope.

**Comment by f00b4r0** [comment]
@vit9696 ok, I'll take your word for it. The fact that it works flawlessly in Clover, the fact that it seems to be fairly random from boot to boot (a telltale sign of a race) and the fact that I could reproduce the problem on two completely different builds made me think this could actually be a bug, especially since I understand OC is aiming to be a superior alternative.
As far as configuration is concerned, I followed the guide on Dortania to the letter and that really is the only problem I have with OC so far. Thanks for taking the time still, I wish I could have been more helpful.

**Comment by vit9696** [comment]
Clover has a lot of quirks, which are not anyhow user-configurable, so I am afraid, even if it does anything special, we will never know. What I can be fairly definite about, is that OpenCore ACPI and kext injection is reliable, and OpenCore does not itself deal with USB hardware aside from ReleaseUsbOwnership quirk. For this reason, even if there is any problem and it is not the configuration, it will be missing functionality rather than a bug.

Yet I have to repeat that it really is unlikely as we have quite a number of ASUS Z370 boards here and they all work more or less fine. Carefully review your BIOS preferences and config.plist. Dortania is not a bad source, but just keep in mind that it is not a magic bullet, as it does not cover all the possible tricks you may run into with a particular board.

**Comment by f00b4r0** [comment]
Purely FYI, as I tried 0.6.1 I did recheck the full BIOS config as well as tried ExitBootServicesDelay and ReleaseUsbOwnership to no avail. I even removed the SSDT-UIAC/USBInjectAll in favor of USBmap, no change. USB power just doesn't want to work with OC.

Out of curiosity, when you say you "have quite a number of ASUS Z370 boards and they all work more or less fine", does that include successful report of USB high power working? In fact USB works, it's just the high current delivery that doesn't, and I suspect most users won't even notice...

[I should also mention in case it matters that I use a custom DSDT as waking up from sleep on this board would storm kernel_task due to spurious gpe6f interrupts. That problem isn't specific to OC, and removing the _L6F method in the DSDT solved the problem.]

I'll stop making noise now :)
